src/inputs/ChipSelectField.tsx

```diff
@@ -142,11 +149,19 @@ export function ChipSelectField<O, V extends Value>(
       if (selectedItem) {
         onSelect(key as V, selectedItem);
       }
+      // Per UX, when an option is selected then we want to call our `onBlur` callback and remove the focus styles. The field _is_ still in focus but that is only to retain tab position in the DOM.
+      // We cannot simply call `buttonRef.current.blur()` here because `state.isOpen === true` and we keep the visualFocus shown when the menu is open.
+      setVisualFocus(false);
+      maybeCall(onBlur);
     },
     onOpenChange: (isOpen) => {
-      if (!isOpen && buttonRef.current) {
-        // When closing reset the focus to the button element.
-        buttonRef.current.focus();
+      if (!isOpen) {
+        // When closing, reset the focus to the button element. This is to retain "tab position" in the document, allowing hte user to hit "Tab" and move to the next tabbable element.
+        // If the menu closed due to a user selecting an option, then the field will not visually appear focused.
+        buttonRef.current?.focus();
+      } else {
+        // If opened, set visual focus to true. It is possible to be in a state where the browser focus is on this element, but we are not "visually" focused (see `onSelectionChange`). If the user opens the menu again, we should trigger the visual focus.
+        setVisualFocus(true);
```

Comment:
Just curious, but why does `SelectField` not need this same "separate `visualFocus`" tracking? I would have assumed the behavior between the two + the impl of "focus briefly moves to a menu" would be similar.

Reply:
SelectField retains the actual focus in the DOM because we still allow the user to type in the input to filter the options. Under the hood it actually uses `useComboBox` which that different focus behavior vs `useSelect`.